### PR TITLE
JATS writer: support author affiliations

### PR DESCRIPTION
**Pull request by tarleb**
Closes: #6687

### CODE REVIEW

**Comment by hubgit** [comment]
@tarleb I think this looks good. 

Will there be a corresponding addition to the JATS reader?

File: data/templates/affiliations.jats [hunk 1]
```diff
@@ -0,0 +1,35 @@
+$--
+$-- Affiliations
+$--
+$for(affiliation)$
+<aff id="aff-$affiliation.id$">
+$-- wrap affiliation if it has a known institution identifier
```
Comment (hubgit) [inline data/templates/affiliations.jats:6]: Should the `<institution-wrap>` opening line be below this comment, before all the `institution` elements?

Also it looks like `<institution-wrap>` is always added, without testing for the presence of an identifier (which is fine, just disagrees with the comment).

File: data/templates/affiliations.jats [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+$if(affiliation.department)$
+<institution content-type="dept">$affiliation.department$</institution>
+$endif$
+<institution-wrap>
+$if(affiliation.organization)$
+<institution>$affiliation.organization$</institution>
+$else$
+<institution>$affiliation.name$</institution>
+$endif$
+$if(affiliation.isni)$
```
Comment (hubgit) [inline data/templates/affiliations.jats:19]: Might it be more future-proof to leave the institution identifiers open and loop over an array of objects with `id` and `id-type` (and perhaps `vocab` and  `vocab-identifier`) properties?

**Comment by tarleb** [comment]
Thanks! I plan to modify the reader eventually, but it will take a good bit longer.

File: data/templates/affiliations.jats (hunk 1 above)
Comment (tarleb) [reply data/templates/affiliations.jats:6]: I thought about doing that. The reason is that `<institution-wrap>`, IIUC, cannot contain PCDATA, so we cannot use commas to separate the different `<institute>` elements as done in the current version.

The comment is indeed outdated, thanks for pointing that out.

File: data/templates/affiliations.jats (hunk 2 above)
Comment (tarleb) [reply data/templates/affiliations.jats:19]: This is the *status quo* for all identifiers right now, including orcid, pmid, pmcid, etc. The current variant is consistent with that. We can add the generic variant on top of that.

**Pull request merged**